ngraph/frontend/ir/src/ir_deserializer.cpp

```diff
@@ -0,0 +1,745 @@
+        if (opName.find(node_param.name) != opName.end() && node_param.type != "Result")
+            IE_THROW() << "Invalid IR! " << node_param.name << " name is not unique!";
+        opName.insert(node_param.name);
+        params[node_param.layerId] = {node, node_param};
+        if (node_param.type == "Result" || node_param.type == "Assign") {
+            outputs.push_back(node_param.layerId);
+        }
+    }
+
+    std::map<size_t /*to-layer-id*/, std::vector<edge>> edges;
+    std::map<size_t, std::shared_ptr<ngraph::Node>> id_to_node;
+
+    // Read all edges and store them for further usage
+    FOREACH_CHILD (_ec, root.child("edges"), "edge") {
+        size_t fromLayer = XMLParseUtils::GetUIntAttr(_ec, "from-layer");
+        size_t fromPort = XMLParseUtils::GetUIntAttr(_ec, "from-port");
+        size_t toLayer = XMLParseUtils::GetUIntAttr(_ec, "to-layer");
+        size_t toPort = XMLParseUtils::GetUIntAttr(_ec, "to-port");
+        edges[toLayer].push_back({fromLayer, fromPort, toPort});
+    }
+
+    // Run DFS starting from outputs to get nodes topological order
+    std::set<size_t> used;
+    std::vector<size_t> order;
+    std::function<void(size_t)> dfs = [&edges, &order, &used, &dfs](const size_t id) {
+        if (used.count(id))
+            return;
+        used.insert(id);
+        for (auto& edge : edges[id]) {
+            dfs(edge.fromLayerId);
+        }
+        order.push_back(id);
+    };
+    std::for_each(outputs.begin(), outputs.end(), dfs);
+
+    // OV_ITT_SCOPE_NEXT(FIRST_INFERENCE, taskChain, "ConstructNgraphNodes");
+
+    FunctionNodes func_nodes;
+
+    std::map<std::string, std::shared_ptr<ngraph::Node>> variable_id_to_read_value;
+
+    //  Following topological order create nGraph operations
+    for (auto& layer_id : order) {
+        auto& p = params[layer_id];
+        const auto& edgeIt = edges.find(layer_id);
+        if (edgeIt == edges.end())
+            continue;
+        ngraph::OutputVector inputs(edgeIt->second.size());
+        for (auto& e : edgeIt->second) {
+            auto input_node = id_to_node[e.fromLayerId];
+            if (!input_node) {
+                IE_THROW() << "Attempt to access node " << e.fromLayerId << " that not in graph.";
+            }
+            auto& p_output = params[e.fromLayerId].params;
+            size_t const realInputPortId = p.params.getRealInputPortId(e.toPortId);
+            if (realInputPortId >= inputs.size())
+                IE_THROW() << p.params.type << " layer " << p.params.name << " with id: " << p.params.layerId
+                           << " is inconsistent!";
+            inputs[realInputPortId] = input_node->output(p_output.getRealOutputPortId(e.fromPortId));
+        }
+
+        auto node = createNode(inputs, p.xml, weights, p.params);
+        id_to_node[layer_id] = node;
+
+        // Check that output shape after nGraph node validation the same as in IR
+        // because IR always right!
+        // Temporary disabled!
```

Comment:
should this be still disabled?

Reply:
Sure. This code was marked as changed because I've split the giant file with all code to the smaller files. So all this deserialization engine was copy-pasted as is. The only change here is rt_info support.